Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/API/JoyfillAPI.swift

```diff
@@ -0,0 +1,358 @@
+                    xCoordinates.append(graphXCoordinateSubArray)
+                    yCoordinates.append(graphYCoordianteSubArray)
+                }
+            }
+            if componentTypeValue == "image" {
+                for k in 0..<valueElements.count {
+                    if let imageURL = URL(string: valueElements[k].url ?? "") {
+                        getImageFromURL(url: imageURL) { image in
+                            if let image = image {
+                                pickedImg.append(image)
+                                DispatchQueue.main.async {
+                                    componentTableView.reloadData()
+                                }
+                            } else {
+                                print("Failed to download image.")
+                            }
+                        }
+                    }
+                }
+            }
+            
+        case .integer(let integer):
+            if componentTypeValue == "number" {
+                numberFieldString = integer
+            }
+        }
+    }
+}
+
+// Function to get the option values
+func getOptionValuesFromPrimaryView(i: Int, j: Int) {
+    optionCount = joyFillStruct?.fields?[j].options?.count ?? 0
+    for n in 0..<optionCount {
+        if componentTypeValue == "dropdown" {
+            dropdownOptions.append(joyFillStruct?.fields?[j].options?[n].value ?? "")
+        }
+        if componentTypeValue == "multiSelect" {
+            multiSelectOptions.append(joyFillStruct?.fields?[j].options?[n].value ?? "")
+        }
+    }
+}
+
+// MARK: - Function for AlternateView(MobileView)
+// Function to get input values
+func getInputValuesFromMobileView(i: Int, j: Int) {
+    if let value = joyFillStruct?.fields?[j].value {
+        switch value {
+        case .string(let string):
+            if componentTypeValue == "textarea" {
+                textAreaString = string
+            }
+            if componentTypeValue == "block" {
+                blockFieldString = string
+                blockTextSize = joyFillStruct?.files?[0].views?[0].pages?[i].fieldPositions?[j].fontSize ?? 18
+                blockTextStyle = joyFillStruct?.files?[0].views?[0].pages?[i].fieldPositions?[j].fontStyle ?? ""
+                blockTextWeight = joyFillStruct?.files?[0].views?[0].pages?[i].fieldPositions?[j].fontWeight ?? ""
+                blockTextColor = joyFillStruct?.files?[0].views?[0].pages?[i].fieldPositions?[j].fontColor ?? "#000000"
+                blockTextAlignment = joyFillStruct?.files?[0].views?[0].pages?[i].fieldPositions?[j].textAlign ?? "left"
+                DispatchQueue.main.async {
+                    componentTableView.reloadData()
+                }
+            }
+            if componentTypeValue == "text" {
+                textFieldString = string
+            }
+            
+        case .valueElementArray(let valueElements):
+            if componentTypeValue == "chart" {
```

Comment:
Looks like this is happening in multiple places here. What is the difference between this one and the one above?

We should not be doing data mapping in multiple places.

Reply:
My apologize for this as I was trying something else based on the functionality like to keep the separate code mobile view and primary view. You're right and I've changes it now